python/cudf_polars/tests/experimental/test_unique.py

```diff
@@ -45,7 +45,19 @@ def test_unique(df, keep, subset, maintain_order, cardinality):
         q = q.select(*(pl.col(col) for col in subset))
         check_row_order = False
 
-    assert_gpu_result_equal(q, engine=engine, check_row_order=check_row_order)
+    is_cardinality0 = cardinality == {}
+
+    should_warn = (maintain_order and (not is_cardinality0 or keep == "none")) or (
+        not maintain_order and (not is_cardinality0) and keep in {"first", "last"}
+    )
+
+    if should_warn:
```

Comment:
Not a problem here since it's relatively simple, but if you want to avoid repeating the `assert_gpu_result_equal(...)` you can put just the context manager definition in the conditional:

```
if should_warn:
    context = pytest.warns(UserWarning, match=...)
else:
    context = contextlib.nullcontext

with context:
    assert_gpu_result_equal(q, engine=engine, check_row_order=check_row_order)
```